packages/editor/src/lib/utils/shapeNavigation.ts

```diff
@@ -0,0 +1,199 @@
+import { TLShape, TLShapeId } from '@tldraw/tlschema'
+import { Editor } from '../editor/Editor'
+import { Vec } from '../primitives/Vec'
+
+const SHALLOW_ANGLE = 20
+const ROW_THRESHOLD = 100
```

Comment:
should these be options??? idk

Reply:
maybe? i think esp. if this is going in the Select tool that maybe it isn't, at least in this first iteration.

Reply:
moved the logic into the Select Tool in the end